clients/src/main/java/org/apache/kafka/common/internals/InvalidProducerEpochException.java

```diff
@@ -16,14 +16,15 @@
  */
 package org.apache.kafka.common.internals;
 
-import org.apache.kafka.common.errors.RetriableException;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.errors.ApiException;
 
 /**
  * This exception indicates that the produce request sent to the partition leader
- * contains a non-matching producer epoch. When encountering this exception, the ongoing transaction
- * will be aborted and can be retried.
+ * contains a non-matching producer epoch. When encountering this exception, user should abort the ongoing transaction
+ * by calling {@link KafkaProducer#abortTransaction()} to reinitialize the producer state.
  */
-public class InvalidProducerEpochException extends RetriableException {
```

Comment:
If we are going to expose this exception, then 1) it cannot be in the internals package, 2) public API javadocs should include this exception, also explains what users should do when getting this error. 3) KIP-588 should include this since it is now a publicly throwable exception.